test/integration/src/test/java/io/pravega/test/integration/ReadTest.java

```diff
@@ -163,9 +163,14 @@ public void testReceivingReadCall() throws Exception {
             SegmentRead result = (SegmentRead) AppendTest.sendRequest(channel, new ReadSegment(segmentName, actual.writerIndex(), 10000, "", 1L));
             assertEquals(segmentName, result.getSegment());
             assertEquals(result.getOffset(), actual.writerIndex());
-            assertTrue(result.isAtTail());
             assertFalse(result.isEndOfSegment());
             actual.writeBytes(result.getData());
+            if (actual.writerIndex() < actual.capacity()) {
```

Comment:
Looks like you could take advantage of the `if` block below to collapse both the assertion logic and the sleep, right?

Reply:
improved the code.